pallets/avn-anchor/src/lib.rs

```diff
@@ -175,21 +185,21 @@ pub mod pallet {
                 Error::<T>::HandlerAlreadyRegistered
             );
 
-            ChainHandlers::<T>::try_mutate(&old_handler, |maybe_chain_data| -> DispatchResult {
-                let chain_data = maybe_chain_data.take().ok_or(Error::<T>::ChainNotRegistered)?;
-                ChainHandlers::<T>::insert(&new_handler, chain_data.clone());
+            ChainHandlers::<T>::try_mutate(&old_handler, |maybe_chain_id| -> DispatchResult {
```

Comment:
This looks a bit strange to me. I would have expected something like:
```
ensure!(
                ChainHandlers::<T>::contains_key(&old_handler),
                Error::<T>::UnauthorizedHandler
);
let chain_id = ChainHandlers::<T>::get(&old_handler);
let chain_data =  ChainData::<T>::get(chain_id).ok_or(Error::<T>::NoChainDataAvailable)?;
ChainHandlers::<T>::insert(&new_handler, chain_id);
ChainHandlers::<T>::remove(&old_handler, chain_id);
```

IMO Inserting a new record while mutating the same underlying storage feels a bit odd to me.